hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/supervisor/workload/SubscriptionAssignmentCache.java

```diff
@@ -60,7 +61,7 @@ public SubscriptionAssignmentCache(CuratorFramework curator,
         this.pathSerializer = new SubscriptionAssignmentPathSerializer(basePath, AUTO_ASSIGNED_MARKER);
         ThreadFactory threadFactory = new ThreadFactoryBuilder().setNameFormat("subscription-assignment-cache-%d").build();
         this.cache = new HierarchicalCache(
-                curator, Executors.newSingleThreadScheduledExecutor(threadFactory), basePath, 2, Collections.emptyList()
+                curator, Executors.newSingleThreadScheduledExecutor(threadFactory), basePath, 2, emptyList(), false
```

Comment:
IMO `HierarchicalCacheBuidler` should be introduced.

Reply:
Thanks for the hint, although as you mentioned this is big PR, so I would like not to introduce further changes not related to the scope of the issue which is fix of integration tests.